src/main/java/org/elasticsearch/index/analysis/SkroutzGreekStemmer.java

```diff
@@ -62,16 +68,23 @@
 public class SkroutzGreekStemmer {
   public final static String DEFAULT_STOPWORD_FILE = "stopwords.txt";
   private final CharArraySet stopwords;
+  private Map<String,char[]> step_0_exceptions;
 
   public SkroutzGreekStemmer(final CharArraySet stopwords) {
     this.stopwords = stopwords;
+    this.step_0_exceptions = new HashMap(0);
```

Comment:
(passing 0 capacity fallbacks to 1) . Even if this is being done for efficiency you have to be sure that it is not gonna be used later on. The reason is that setting its capacity to 0 combined with the default loadFactor which is 0.75, means that the hash map will have to resize internally everytime a new object is added ( resizes when currentSize >= (capacity*loadfactor) ).